src/lib/validators.js

```diff
@@ -132,3 +133,26 @@ export const validateUnique = arr => {
 
 export const validateChild = ourShip => ship =>
   ourShip !== ob.sein(ship) && `This point is not a child of ${ourShip}.`;
+
+export const validatePsbt = base64 => {
+  try {
+    bitcoin.Psbt.fromBase64(base64);
+  } catch (e) {
+    return 'Invalid Partially Signed Bitcoin Transaction';
+  }
+};
+
+export const validateSignablePsbt = hd => base64 => {
+    const psbt = bitcoin.Psbt.fromBase64(base64);
+    return !psbt.data.inputs.some((input, idx) => {
+      try {
+        //  removing already derived part, eg 'm/84'/0'/0'/0/0' becomes '0/0'
+        const path = input.bip32Derivation[0].path.substring(12);
+        const prv = hd.derivePath(path).privateKey;
+        psbt.signInput(idx, bitcoin.ECPair.fromPrivateKey(prv));
+        return true;
+      } catch (e) {
+        return false;
```

Comment:
To make sure I understand correctly: this would crash if the input was, uh, partially signed with a different private key?

Reply:
The point is to allow signing if one or more of the inputs are able to be signed. I noticed that the validation here supported it, but the actual signing in `Bitcoin.js` didn't, so I added 61d0d8e.